File: frontend/numerize_me_app/lib/models/questionnaire.dart
class BinaryQuestion {
  final int id;
  final String text;
  final String optionA;
  final String optionB;

  BinaryQuestion({
    required this.id,
    required this.text,
    required this.optionA,
    required this.optionB,
  });

  factory BinaryQuestion.fromJson(Map<String, dynamic> json) {
    return BinaryQuestion(
      id: json['id'],
      text: json['text'],
      optionA: json['option_a'],
      optionB: json['option_b'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'text': text,
      'option_a': optionA,
      'option_b': optionB,
    };
  }
}

class Questionnaire {
  final int id;
  final String title;
  final String description;
  final String category;
  final int weight;
  final List<BinaryQuestion> questions;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final bool isCompleted;

  Questionnaire({
    required this.id,
    required this.title,
    required this.description,
    required this.category,
    required this.weight,
    required this.questions,
    required this.createdAt,
    this.updatedAt,
    this.isCompleted = false,
  });

  factory Questionnaire.fromJson(Map<String, dynamic> json) {
    var questionsList = json['questions'] as List;
    List<BinaryQuestion> questions = questionsList
        .map((q) => BinaryQuestion.fromJson(q))
        .toList();

    return Questionnaire(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      category: json['category'],
      weight: json['weight'],
      questions: questions,
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: json['updated_at'] != null 
          ? DateTime.parse(json['updated_at']) 
          : null,
      isCompleted: json['is_completed'] ?? false,
    );
  }
}

class BinaryAnswer {
  final int questionId;
  final String chosenOption;

  BinaryAnswer({
    required this.questionId,
    required this.chosenOption,
  });

  Map<String, dynamic> toJson() {
    return {
      'question_id': questionId,
      'chosen_option': chosenOption,
    };
  }
}

File: backend/app/schemas/questionnaire.py
from typing import List, Optional, Dict, Union, Any
from pydantic import BaseModel, Field
from datetime import datetime
from enum import Enum

class BinaryQuestion(BaseModel):
    """Question binaire simplifiée : choix entre 2 options seulement"""
    id: int
    text: str = Field(..., description="Le texte de la question (ex: 'Pizza vs Sushis')")
    option_a: str = Field(..., description="Première option (ex: 'Pizza')")
    option_b: str = Field(..., description="Deuxième option (ex: 'Sushis')")

class QuestionnaireBase(BaseModel):
    title: str
    description: str
    category: str
    weight: int = 1
    questions: List[BinaryQuestion] = Field(..., description="Liste des questions binaires")

class QuestionnaireCreate(QuestionnaireBase):
    pass

class QuestionnaireResponse(BaseModel):
    id: int
    title: str
    description: str
    category: str
    weight: int
    questions: List[BinaryQuestion]
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True

class BinaryAnswer(BaseModel):
    """Réponse à une question binaire"""
    question_id: int
    chosen_option: str = Field(..., description="L'option choisie (option_a ou option_b)")

class UserResponseBase(BaseModel):
    questionnaire_id: int
    answers: List[BinaryAnswer] = Field(..., description="Liste des réponses binaires")

class UserResponseCreate(UserResponseBase):
    pass

class UserResponseResponse(BaseModel):
    id: int
    user_id: int
    questionnaire_id: int
    answers: List[BinaryAnswer]
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True
